app/src/main/java/com/project/ianime/repository/AnimeDataRepositoryImpl.kt

```diff
@@ -0,0 +1,64 @@
+package com.project.ianime.repository
+
+import com.project.ianime.api.AnimeService
+import com.project.ianime.api.data.AnimeGalleryItem
+import com.project.ianime.api.error.*
+import com.project.ianime.api.model.AnimeApiModel
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.schedulers.Schedulers
+import retrofit2.HttpException
+import javax.inject.Inject
+
+class AnimeDataRepositoryImpl @Inject constructor(private val animeService: AnimeService) :
+    AnimeDataRepository {
+
+    override fun getAnimeListFromNetwork(): Single<List<AnimeApiModel>> {
+        return animeService.getAnimeList()
+            .subscribeOn(Schedulers.io())
+            .map { it.animeList }
+            .onErrorResumeNext { exception ->
+                if (exception is HttpException) {
+                    exception.code().let {
+                        when {
+                            it == HttpStatus.BAD_REQUEST -> {
+                                return@onErrorResumeNext Single.error(
+                                    BadRequestException(exception.message())
+                                )
+                            }
+                            it == HttpStatus.UNAUTHORIZED -> {
+                                return@onErrorResumeNext Single.error(
+                                    UnauthorizedException(exception.message())
+                                )
+                            }
+                            it == HttpStatus.NOT_FOUND -> {
+                                return@onErrorResumeNext Single.error(
+                                    NotFoundException(exception.message())
+                                )
+                            }
+                            it >= 500 -> {
```

Comment:
add content too large err

Reply:
content too large is only for post or put request right? get request only gets the url?

Reply:
right, only for post, patch, put, get gets url only for now